### PR TITLE
Use BenchmarkTools for timing

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/benchmark.jl [hunk 1]
```diff
@@ -333,28 +346,36 @@ function WorkPrecisionSet(prob::AbstractRODEProblem,abstols,reltols,setups,test_
...
+          @belapsed(solve($prob,$(setups[k][:alg]);
+                          $kwargs...,abstol=$(abstols[j]),
+                          reltol=$(reltols[j]),dt=$(setups[k][:dts][j]),
+                          timeseries_errors=false,
+                          dense_errors = false), seconds = 0.1)
         end
+      catch
+        # sometimes BenchmarkTools errors with
+        # `ERROR: syntax: function argument and static parameter names must be distinct`
+        # so, we are catching that error and try a few times.
```
Comment (ChrisRackauckas) [inline src/benchmark.jl]: why does it do that?

File: src/benchmark.jl [hunk 2]
```diff
@@ -333,28 +346,36 @@ function WorkPrecisionSet(prob::AbstractRODEProblem,abstols,reltols,setups,test_
...
                 timeseries_errors=false,
-                dense_errors = false)
+                dense_errors = false), seconds = 0.1)
         else
-          sol = solve(prob,setups[k][:alg];
-                kwargs...,abstol=abstols[j],
-                reltol=reltols[j],dt=setups[k][:dts][j],
-                timeseries_errors=false,
-                dense_errors = false)
+          @belapsed(solve($prob,$(setups[k][:alg]);
```
Comment (ChrisRackauckas) [inline src/benchmark.jl]: minimum doesn't really make sense for stochastic

**Comment by ChrisRackauckas** [comment]
I'd like to keep the numruns option, where if it's set then you get the average time from the runs. That way we have an easy way to double check that BenchmarkTools is really giving a stable enough answer.

File: src/benchmark.jl (hunk 1 above)
Comment (YingboMa) [reply src/benchmark.jl]: I am not sure. It looks like something relate to macros.

Comment (YingboMa) [reply src/benchmark.jl]: https://github.com/JuliaCI/BenchmarkTools.jl/issues/135 upstream issue

**Pull request merged**